builtin/plugins/chainconfig/chainconfig.go

```diff
@@ -350,6 +368,119 @@ func EnableFeatures(ctx contract.Context, blockHeight, buildNumber uint64) ([]*F
 	return enabledFeatures, nil
 }
 
+// UpdateConfig returns a list of settings that are supported by majority of validators
+func UpdateConfig(ctx contract.Context, buildNumber uint64) ([]*Setting, error) {
+	params, err := getParams(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	settingsRange := ctx.Range([]byte(settingPrefix))
+	settings := make([]*Setting, 0)
+
+	validatorsInfo, err := listValidatorsInfo(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, m := range settingsRange {
+		var setting Setting
+		if err := proto.Unmarshal(m.Value, &setting); err != nil {
+			return nil, errors.Wrapf(err, "unmarshal Setting %s", string(m.Key))
+		}
+
+		supportedValidator := 0
+		for _, validatorInfo := range validatorsInfo {
+			if validatorInfo.BuildNumber >= setting.BuildNumber {
+				supportedValidator++
+			}
+		}
+		// Return this setting, if the number of validators that support this setting
+		// has not reached the vote threshold
+		if len(validatorsInfo) > 0 && uint64((supportedValidator*100)/len(validatorsInfo)) >= params.VoteThreshold {
+			if buildNumber < setting.BuildNumber {
+				return nil, ErrConfigNotSupported
+			}
+			settings = append(settings, &setting)
+		}
+	}
+
+	return settings, nil
+}
+
+// RemoveSetting removes setting in the contract
+func RemoveSetting(ctx contract.Context, name string) {
+	ctx.Delete(settingKey(name))
+}
+
+// GetSetting returns info about a specific cfg setting.
+func (c *ChainConfig) GetSetting(ctx contract.StaticContext, req *GetSettingRequest) (*GetSettingResponse, error) {
```

Comment:
Probably don't need this one, ListSettings/ListPendingActions should be enough

Reply:
Removed